iml-wasm-components/iml-input/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+extern crate combine;
+extern crate seed;
+
+use combine::stream::easy::{Error, Errors, Info};
+use seed::{div, input, prelude::*};
+
+use iml_tooltip::TooltipPlacement;
+
+pub struct Model<T> {
+    pub value: Result<T, String>,
+    pub tooltip: iml_tooltip::Model,
+}
+
+impl<T> Default for Model<T> {
+    fn default() -> Self {
+        Self {
+            value: Err(String::new()),
+            tooltip: iml_tooltip::Model {
+                error_tooltip: true,
+                open: true,
+                placement: TooltipPlacement::Bottom,
+                ..Default::default()
+            },
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum Msg {
+    FlipTooltip,
+    InputChanged(String),
+}
+
+pub type ParserFn<T> = fn(_: &str) -> Result<T, Errors<char, &str, usize>>;
+
+fn format_err(err: &Errors<char, &str, usize>) -> String {
+    let mut un = vec![];
+    let mut ex = vec![];
+
+    fn format_info(i: &Info<char, &str>) -> String {
+        match i {
+            Info::Token(t) => {
+                if *t != '"' {
+                    format!("\"{}\"", t)
+                } else {
+                    format!("{}", t)
+                }
+            }
+            Info::Static(s) => String::from(*s),
+            Info::Owned(s) => String::from(s),
+            Info::Range(r) => String::from(*r),
+        }
+    }
+
+    for x in &err.errors {
+        match x {
```

Comment:
This is not totally covering the error types. Do we want to do something with those?

Reply:
The other two is a "generic" message and "other, but Combine [never generates them](https://docs.rs/combine/4.0.0-beta.1/combine/stream/easy/struct.Errors.html), and their use is not clear to decide how to show them.